app/presenters/tree_builder.rb

```diff
@@ -247,14 +247,14 @@ def set_locals_for_render
   # :add_root               # If true, put a root node at the top
   # :full_ids               # stack parent id on top of each node id
   def x_build_dynatree(options)
-    children = x_get_tree_objects(nil, options, nil, [])
+    children = x_get_tree_objects(nil, options, [])
 
-    child_nodes = children.each_with_object([]) do |child, acc|
+    child_nodes = children.map do |child|
       # already a node? FIXME: make a class for node
       if child.kind_of?(Hash) && child.key?(:title) && child.key?(:key) && child.key?(:icon)
-        acc << child
+        child
       else
-        acc.concat(x_build_node_dynatree(child, nil, options))
+        x_build_node_dynatree(child, nil, options)
```

Comment:
I believe this is slightly different now because continuous concats produces a different values in the end, then map...

```ruby
%w(a b c).map { |x| ["#{x}1", "#{x}2"] }
# => [["a1", "a2"], ["b1", "b2"], ["c1", "c2"]]
%w(a b c).each_with_object([]) { |x, a| a.concat(["#{x}1", "#{x}2"]) }
# => ["a1", "a2", "b1", "b2", "c1", "c2"]
```